sherpa/astro/io/xstable.py

```diff
@@ -0,0 +1,752 @@
+    """
+    hardmin: float
+    """The minimum value for the parameter."""
+    hardmax: float
+    """The maximum value for the parameter."""
+    softmin: Optional[float] = None
+    """The "soft" minimum. Defaults to hardmin if not given.."""
+    softmax: Optional[float] = None
+    """The "soft" maximum. Defaults to hardmax if not given."""
+
+    def __post_init__(self):
+        """Validate the settings"""
+
+        # Do we need to set up softmin/max?
+        #
+        if self.softmax is None:
+            self.softmax = self.hardmax
+
+        if self.softmin is None:
+            self.softmin = self.hardmin
+
+        # There should probably be a check that the name is not
+        # an invalid Python attribute name.
+        #
+        if self.name.strip() == "":
+            raise ValueError("name can not be empty")
+
+        if self.softmin < self.hardmin:
+            raise ValueError(f"Parameter {self.name} softmin={self.softmin} < hardmin={self.hardmin}")
+
+        if self.softmax > self.hardmax:
+            raise ValueError(f"Parameter {self.name} softmax={self.softmax} > hardmax={self.hardmax}")
+
+        # We allow min == max
+        if self.softmin > self.softmax:
+            raise ValueError(f"Parameter {self.name} softmin={self.softmin} > softmax={self.softmax}")
+
+        if self.initial < self.softmin or self.initial > self.softmax:
+            raise ValueError(f"Parameter {self.name} initial={self.initial} outside softmin={self.softmin} to softmax={self.softmax}")
+
+        if self.delta == 0:
+            raise ValueError("delta can not be 0")
+
+
+@dataclass
+class IntParam(Param):
+    """Represent an interpolated parameter."""
+
+    values: list[float] = field(default_factory=list)
+    """The parameter values used to create the model spectra.
+
+    Is is required that they range from hardmin to hardmax and are in
+    monotonically increasing order (so it can not be an empty list).
+    """
+    loginterp: bool = False
+    """Are the values logarithmically interpolated (True) or linearly (False).
+
+    Defaults to linear interpolation.
+    """
+
+    def __post_init__(self):
+        """Validate the settings"""
+        super().__post_init__()
+
+        if len(self.values) == 0:
+            raise ValueError(f"Parameter {self.name} has no values")
+
+        if self.values[0] != self.hardmin:
+            raise ValueError(f"Parameter {self.name} hardmin={self.hardmin} != first value={self.values[0]}")
+
+        if self.values[-1] != self.hardmax:
+            raise ValueError(f"Parameter {self.name} hardmax={self.hardmax} != last value={self.values[-1]}")
+
+        if np.any(np.diff(self.values) <= 0):
+            raise ValueError(f"Parameter {self.name} values are not monotonically increasing")
+
+
+# Perhaps the HeaderItem and Column types can be used in
+# sherpa.astro.io to pass information to and from the backends, rather
+# than the current system. However, that is for later work.
+#
```

Comment:
We do re-impleent pycrates and astropy.io.pyfits here, though, both of which have classes for header cards, Columns, tables and HDUs. I guess that's the price we pay for having abstraction from the backend. So, passing those in, instead of introducing another layer of abstraction by turning it into dicts and lists to pass to the backend code seems duplication. Also, We might want to move the definitions of these classes into `sherpa.io` and use them inside the other backend functions, too. I think that might actually simplify a lot of the backend code. But I'm OK with placing it here for now and moving it later.

Reply:
I think I refer to the comment on line 246..... ;-)

Reply:
I m using this idea in #1869 but at a different level of abstraction (storing data related to a particular "block" of data). So we could be just abstracting ourselves to oblivion (also that code doesn't take advantage of the "HeaderCard" interface provided here when maybe it should).

Reply:
I also have moved these types into a separate module there - `io_types` - but that's not worth doing here.